Source/iOS/OIDExternalUserAgentCatalyst.h

```diff
@@ -0,0 +1,48 @@
+/*! @file OIDExternalUserAgentCatalyst.h
+   @brief AppAuth iOS SDK
+   @copyright
+       Copyright 2019 The AppAuth Authors. All Rights Reserved.
+   @copydetails
+       Licensed under the Apache License, Version 2.0 (the "License");
+       you may not use this file except in compliance with the License.
+       You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+       Unless required by applicable law or agreed to in writing, software
+       distributed under the License is distributed on an "AS IS" BASIS,
+       WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+       See the License for the specific language governing permissions and
+       limitations under the License.
+*/
+
+#import <UIKit/UIKit.h>
+
+#import "OIDExternalUserAgent.h"
+
+#if TARGET_OS_MACCATALYST
+
+NS_ASSUME_NONNULL_BEGIN
+
+/*! @brief A Catalyst specific external user-agent that uses `ASWebAuthenticationSession` to
+       present the request.
+*/
+@interface OIDExternalUserAgentCatalyst : NSObject<OIDExternalUserAgent>
+
+- (nullable instancetype)init API_AVAILABLE(ios(11))
+__deprecated_msg("This method will not work on iOS 13, use "
```

Comment:
Do we need this deprecated method, since this class is new?

Reply:
In `OIDAuthState+IOS.m` we have a method which doesn't require a presenting view controller, that is the reason why I kept that init until we remove the ability to not provide a view controller.

But, since this class is new, I can remove this deprecated method and assert in `OIDAuthState+IOS.m` instead for Catalyst, what I prefer.

Reply:
Sounds good.

I'd probably go with the `#if !TARGET_OS_MACCATALYST`. Since the Catalyst AppAuth API is "new" we're not breaking any backwards comparability of the existing API surface by removing it. Either way is reasonable.

Reply:
Done. I marked this convenience method with `API_UNAVAILABLE(macCatalyst)`.